src/Abstractions/NexusMods.Abstractions.Settings/ISettingsPropertyUIDescriptor.cs

```diff
@@ -0,0 +1,43 @@
+using JetBrains.Annotations;
+
+namespace NexusMods.Abstractions.Settings;
+
+/// <summary>
+/// Represents a descriptor for settings properties exposed to the UI.
+/// </summary>
+[PublicAPI]
+public interface ISettingsPropertyUIDescriptor
+{
+    /// <summary>
+    /// Gets the Section ID.
+    /// </summary>
+    public SectionId SectionId { get; }
+
+    /// <summary>
+    /// Gets the display name.
+    /// </summary>
+    public string DisplayName { get; }
+
+    /// <summary>
+    /// Gets the markdown description.
+    /// </summary>
+    public string Description { get; }
+
+    /// <summary>
+    /// Gets whether changing the property requires a restart.
+    /// </summary>
+    public bool RequiresRestart { get; }
+
+    /// <summary>
+    /// Gets the optional custom restart message.
+    /// </summary>
+    /// <remarks>
+    /// If this is <c>null</c>, use a default message.
+    /// </remarks>
+    public string? RestartMessage { get; }
```

Comment:
I had not thought of us needing this, do you have a use case in mind?

Reply:
Do we need one? Any case were the generic one would not suffice?

Reply:
I don't know? I'm sure we can think of something. Doesn't really matter if it goes unused or not.

Reply:
👍 I didn't plan to add such functionality in the UI for now